ruby-challenges.rb

```diff
@@ -19,16 +30,68 @@
 nums_array2 = [25, 17, 47, 11]
 # Expected output: 100
 
+# Create a method called num_sum and set parameter as array
+# Inside the method use the sum method on the array parameter to return the sum total of the numbers in the array
+# Call the method and pass in arguments
+def num_sum(array)
+    p array.sum
+end
 
+num_sum(nums_array1)
+num_sum(nums_array2)
 
 # --------------------3a) Create a class called Bike that is initialized with a model, wheels, and current_speed. The default number of wheels is 2. The current_speed should start at 0. Create a bike_info method that returns a sentence with all the data from the bike object.
 
 # Expected output example: 'The Trek bike has 2 wheels and is going 0 mph.'
 
+# Create a class called Bike
+# Initialize model, wheels, and current_speed, where model can have variable input but wheels is set to 2 and current_speed is set to 0
+# Create a method called bike_info which uses string interpolation to print the string "The #{@model} bike has #{@wheels} wheels and is going #{@current_speed} mph."
+# Create a method called pedal_faster which increases current_speed by the number passed in to parameter num
+# Create a method called brake which decreases current_speed by the number passed in to parameter num
+# The number cannot go below zero so set a variable called decrease_speed equal to current_speed minus num
+# Create a conditional which specifies that if decrease_speed is less than zero, current_speed should equal zero
+# Else current_speed equals decrease_speed
+# Create a new bike called my_bike and set model to "Trek"
+# Test methods
+class Bike
+    def initialize(model)
+        @model = model
+        @wheels = 2
+        @current_speed = 0
+    end
+
+    def bike_info
+        "The #{@model} bike has #{@wheels} wheels and is going #{@current_speed} mph."
+    end
+
+    def pedal_faster(num)
+        @current_speed += num
+    end
 
+    def brake(num)
+        decrease_speed = @current_speed - num
+        if decrease_speed < 0
+            @current_speed = 0
+        else
+            @current_speed = decrease_speed
+        end
+    end
+end
```

Comment:
This looks excellent!  For a stretch, look into ternary operators in Ruby for your conditional statements.